### DISCUSSION TITLE
RTMP to RTP server directly

### DISCUSSION THREAD

**Discussion by RomanGalo4kin** [bluenviron/mediamtx#593] [General]
Is it possible to proxy video & audio tracks to RTP server directly? I mean RTP chunks.

Because, I see that here "rtp" lib from "pion" is used. But in the RMTP specification, I do not find any word about "RTP". So, why this library was used?

https://github.com/aler9/rtsp-simple-server/blob/main/internal/core/rtmp_conn.go

**Comment by aler9** [comment]
RTP is used as part of the RTSP implementation, not the RTMP one.

The difference between RTP and RTSP is that the latter adds an initial handshake that allows to send the track list and the codec list to the client, then the transmission works almost in the same way, by using RTP packets. There's not a difference in performance.

You can transmit a RTMP stream from a camera to the server, and then proxy it by using RTSP, conversion is transparent.

[a comment by an automated account is not shown]
